public/resources/remix/index.js

```diff
@@ -6,122 +6,126 @@
+      if (isTouchDevice) {
+        detailsBar.classList.add("touch-mode");
+      } else {
+        detailsBar.classList.add("mouse-mode");
+      }
 
-    if(isTouchDevice) {
-      detailsBar.addClass("touch-mode");
-    } else {
-      detailsBar.addClass("mouse-mode");
+      detailsBar.addEventListener("click", function(event) {
+        if (event.target.classList.contains("thimble-button")) {
+          detailsBar.classList.remove("collapsed");
+          event.stopPropagation();
+        }
+      });
+
+      detailsBar.addEventListener("click", function(event) {
+        if (event.target.classList.contains("close-details-bar")) {
+          detailsBar.classList.add("collapsed");
+          event.stopPropagation();
+        }
+      });
+
+      document.querySelector(".mouse-mode").addEventListener("mouseenter", function(){
+        detailsBar.classList.remove("collapsed");
+      });
+
+      document.querySelector(".mouse-mode").addEventListener("mouseleave", function(){
+        detailsBar.classList.add("collapsed");
+      });
     }
-
-    detailsBar.on("click", ".thimble-button",function(){
-      detailsBar.removeClass("collapsed");
-      return false;
-    });
-
-    detailsBar.on("click", ".close-details-bar", function(){
-      detailsBar.addClass("collapsed");
-      return false;
-    });
-
-    $(".mouse-mode").on("mouseenter",function(){
-      detailsBar.removeClass("collapsed");
-    });
-
-    $(".mouse-mode").on("mouseleave",function(){
-      detailsBar.addClass("collapsed");
-    });
   }
 
-  function injectDetailsBar($, metadata, callback) {
-    $.ajax({
-      url: metadata.host + "/projects/remix-bar",
-      cache: false,
-      accepts: "text/html",
-      dataType: "html",
-      data: {
-        host: metadata.host,
-        id: metadata.projectId,
-        title: metadata.projectTitle,
-        author: metadata.projectAuthor,
-        updated: metadata.dateUpdated
+  function injectDetailsBar(metadata, callback) {
+    var xmlhttp = new XMLHttpRequest();
+    xmlhttp.responseType = "text";
+    var url = metadata.host + "/projects/remix-bar";
+    var data = {
+      host: metadata.host,
+      id: metadata.projectId,
+      title: metadata.projectTitle,
+      author: metadata.projectAuthor,
+      updated: metadata.dateUpdated
+    };
+
+    xmlhttp.onreadystatechange = function() {
+      if (xmlhttp.readyState === XMLHttpRequest.DONE ) {
+        var response = xmlhttp.response;
+        var textStatus = xmlhttp.statusText;
+        if (xmlhttp.status === 200) {
+          document.body.insertBefore(response, document.body.firstChild);
+	   callback(null);
+        } else if (xmlhttp.status >= 400) {
+	   callback(textStatus);
+        }
       }
-    })
-    .done(function(response) {
-      $(response).prependTo("body");
-      callback(null, $);
-    })
-    .fail(function(jqXHR, textStatus) {
-      callback(textStatus);
-    });
+    };
+
+    xmlhttp.open("POST", url, true);
```

Comment:
This is absolutely not a `POST` operation: we are *getting* a file from the server, we are not pushing data into the server for storage.